### PR TITLE
[Feature](Planner) Support `SET_VAR` hint in insert statement

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

Currently, when doing partial update using insert statement, we need to set `enable_unique_key_partial_update=true` and always need to set `enable_insert_strict=true`, which may pollute the session variables if the users forget to set them to the original values after executing the statements. So we allow to set variables in insert statement using comment to keep the influence only available in the current statement.

Example:
```sql
insert /*+ SET_VAR(enable_unique_key_partial_update=true, enable_insert_strict = false)*/
into tbl(id,score) values(2,400),(1,200),(4,400);
```

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

File: regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_insert_with_hint.groovy [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                `test` int(11) NULL COMMENT "null test",
+                `dft` int(11) DEFAULT "4321")
+                UNIQUE KEY(`id`) DISTRIBUTED BY HASH(`id`) BUCKETS 1
+                PROPERTIES("replication_num" = "1", "enable_unique_key_merge_on_write" = "true")
+    """
+    sql """insert into ${tableName} values(2, "doris2", 2000, 223, 1),(1, "doris", 1000, 123, 1)"""
+    qt_1 """ select * from ${tableName} order by id; """
+    // partial update using insert stmt in non-strict mode,
+    // existing rows should be updated and new rows should be inserted with unmentioned columns filled with default or null value
+    sql """insert /*+ SET_VAR(enable_unique_key_partial_update=true, enable_insert_strict = false)*/
```
Comment (zhannngchen) [inline regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_insert_with_hint.groovy:40]: need to cover more tests for session variables related with insert into

**Comment by zhannngchen** [comment]
Also need to update documents

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java [hunk 2]
```diff
@@ -117,7 +117,9 @@ private Plan bindOlapTableSink(MatchingContext<UnboundTableSink<Plan>> ctx) {
         Pair<Database, OlapTable> pair = bind(ctx.cascadesContext, sink);
         Database database = pair.first;
         OlapTable table = pair.second;
-        boolean isPartialUpdate = sink.isPartialUpdate() && table.getKeysType() == KeysType.UNIQUE_KEYS;
+        boolean isPartialUpdate = (table.getKeysType() == KeysType.UNIQUE_KEYS
+                && table.getEnableUniqueKeyMergeOnWrite() && (sink.isPartialUpdate()
+                || ctx.cascadesContext.getConnectContext().getSessionVariable().isEnableUniqueKeyPartialUpdate()));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:122]: changes not related with this PR?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java [hunk 3]
```diff
@@ -395,16 +411,21 @@ private static Expression castValue(Expression value, DataType targetType) {
...
     public static TableIf getTargetTable(Plan plan, ConnectContext ctx) {
         UnboundLogicalSink<? extends Plan> unboundTableSink;
-        if (plan instanceof UnboundTableSink) {
-            unboundTableSink = (UnboundTableSink<? extends Plan>) plan;
-        } else if (plan instanceof UnboundHiveTableSink) {
-            unboundTableSink = (UnboundHiveTableSink<? extends Plan>) plan;
-        } else if (plan instanceof UnboundIcebergTableSink) {
-            unboundTableSink = (UnboundIcebergTableSink<? extends Plan>) plan;
+        Plan sinkPlan = plan;
+        if (plan instanceof LogicalStatementHint) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java:415]: too many `if (plan instanceOf LogicalStatementHint) {...} else {...}` in different place. Maybe it is better to save hint in InserIntoTableCommand, and add it back to plan tree before do real plan in `initPlan`. or do `eliminate logical statement` at very begining in InsertIntoTableCommand

**Pull request closed without merging**